src/pages/_nav_bar.tsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import { animated, useSprings } from "@react-spring/web";
+import { MenuAlt2Icon, ArrowRightIcon } from "@heroicons/react/outline";
+
+function NavBarComponent({ customMouseEnter, customMouseLeave }: any) {
+  const bg_colors = ["bg-amber-400", "bg-red-400", "bg-blue-400"];
+  const [_springs, _apis] = useSprings(bg_colors.length + 1, (index) => ({
+    from: {
+      x: 2000,
+      y: 0,
+    },
+  }));
+
+  const menu_values = [
```

Comment:
Use the `useMemo` hook to optimize the `menu_values` array so that they are only recomputed if their dependencies change. This will prevent unnecessary re-renders of the component.